File: autumn-rpc/autumn-rpc-test/src/main/java/com/example/test/TestService.java
package com.example.test;

import cn.imaq.autumn.rpc.server.annotation.AutumnRPCInterface;

import java.rmi.Remote;
import java.rmi.RemoteException;
import java.util.List;

@AutumnRPCInterface
public interface TestService extends Remote {
    String echo(String input) throws RemoteException;

    MyObject testEnum(int num) throws RemoteException;

    String testObject(String str, MyObject object) throws RemoteException;

    String testArray(Object[] arr) throws RemoteException;

    List<MyObject> testList(List<MyObject> list) throws RemoteException;

    void testThrowException(String msg) throws Exception;

    Exception testReturnException(String msg) throws RemoteException;
}

File: autumn-rpc/cross-lang/dotnet/DeepAQ.AutumnRPC.TestApplication/TestNamespace/TestService.cs
using System;
using System.Collections.Generic;

namespace DeepAQ.AutumnRPC.TestApplication.TestNamespace
{
    public interface TestService
    {
        string echo(string input);

        MyObject testEnum(int num);

        string testObject(string str, MyObject o);

        string testArray(object[] arr);

        IList<MyObject> testList(IList<MyObject> list);

        void testThrowException(string msg);

        Exception testReturnException(string msg);
    }
}